test/e2e/apimachinery/garbage_collector.go

```diff
@@ -1102,13 +1102,11 @@ var _ = SIGDescribe("Garbage collector", func() {
 		}
 
 		// Wait 30s and ensure the dependent is not deleted.
-		ginkgo.By("wait for 30 seconds to see if the garbage collector mistakenly deletes the dependent crd")
-		if err := wait.PollWithContext(ctx, 5*time.Second, 30*time.Second+gcInformerResyncRetryTimeout, func(ctx context.Context) (bool, error) {
+		ginkgo.By("wait for 90 seconds to see if the garbage collector mistakenly deletes the dependent crd")
+		gomega.Consistently(func() error {
 			_, err := resourceClient.Get(ctx, dependentName, metav1.GetOptions{})
-			return false, err
-		}); err != nil && !wait.Interrupted(err) {
-			framework.Failf("failed to ensure the dependent is not deleted: %v", err)
-		}
+			return err
+		}, 30*time.Second+gcInformerResyncRetryTimeout, 5*time.Second).Should(gomega.BeNil())
```

Comment:
Have you tried what I suggested? It's a bit simpler and handles retries properly, something that this code here doesn't.

Reply:
I see. Can you add `HandleRetry` (https://github.com/kubernetes/kubernetes/blob/38ed3ef7b7d00353ab81332c5c34e1cb5abec3f2/test/e2e/framework/get.go#L48C9-L48C20)?

Also, please use `gomega.Succeed()`. It provides a better failure message than `BeNil`.

Reply:
Another problem with your `func() error` approach is that it doesn't use the context provided by `gomega.Consistently`. Instead, the anonymous function uses the context from the surrounding test. Please use `func(ctx context.Context) error` instead.

Reply:
Yes, thanks. Is it right then?
```
gomega.Consistently(ctx, framework.HandleRetry[*unstructured.Unstructured](
  func(ctx context.Context) (*unstructured.Unstructured, error) {
  return resourceClient.Get(ctx, dependentName, metav1.GetOptions{})})).WithTimeout(30*time.Second + gcInformerResyncRetryTimeout).WithPolling(5*time.Second).Should(gomega.Succeed())
```